lib/assert.js

```diff
@@ -369,9 +380,143 @@ function isSpecial(obj) {
 }
 
 const typesToCallDeepStrictEqualWith = [
-  isKeyObject, isWeakSet, isWeakMap, Buffer.isBuffer,
+  isKeyObject, isWeakSet, isWeakMap, Buffer.isBuffer, isSharedArrayBuffer,
 ];
 
+function compareMaps(actual, expected, comparedObjects) {
+  if (actual.size !== expected.size) {
+    return false;
+  }
+  const safeIterator = FunctionPrototypeCall(SafeMap.prototype[SymbolIterator], actual);
+
+  comparedObjects ??= new SafeWeakSet();
+
+  for (const { 0: key, 1: val } of safeIterator) {
+    if (!MapPrototypeHas(expected, key)) {
+      return false;
+    }
+    if (!compareBranch(val, MapPrototypeGet(expected, key), comparedObjects)) {
+      return false;
+    }
+  }
+  return true;
+}
+
+function compareArrayBuffers(actual, expected) {
+  let actualView, expectedView, expectedViewLength;
+
+  if (!ArrayBufferIsView(actual)) {
```

Comment:
Can you add some tests where you partially compare an `ArrayBuffer` with a `DataView`? I'd expect you need an additional check otherwise you'd get a `TypeError`
```suggestion
  if (!ArrayBufferIsView(actual)) {
    if (ArrayBufferIsView(expected)) return false;
```

Reply:
@aduh95 you are right, I missed one case! Updated, thanks 😄